### PR TITLE
ADD litmus-agent helm-chart

### PR DESCRIPTION
**Pull request by Vr00mm**
Signed-off-by: Rémi ZIOLKOWSKI <remi.ziolkowski-ext@pole-emploi.fr>

#### What this PR does / why we need it:
This PR add an helm-chart for litmus-agent

!!!!!!
This helm-chart require a docker-image for the pre/post hooks.
IDK where to publish the go source code of that image.
!!!!!!

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] DCO signed
- [] Chart Version bumped
- [X] Variables are documented in the README.md

### CODE REVIEW

**Comment by ksatchit** [comment]
@Vr00mm thanks for adding this! Such images typically go into https://github.com/litmuschaos/test-tools. You could add a dedicated folder for the source! 

(I know the name 'test-tools' ins't entirely appropriate - we can discuss the name :-) )

**Comment by Vr00mm** [comment]
Hello @ksatchit,
Thanx for details.
I created the pull request here -> https://github.com/litmuschaos/test-tools/pull/208

File: charts/litmus-agent/charts/subscriber/templates/hpa.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{{- if .Values.autoscaling.enabled }}
```
Comment (gdsoumya) [inline charts/litmus-agent/charts/subscriber/templates/hpa.yaml:1]: We cannot scale the subscriber currently, there are limitations and edge cases that will break the portal on scaling subscriber.

File: charts/litmus-agent/charts/workflow-controller/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          args:
+            - --configmap
+            - workflow-controller-configmap
+            - --executor-image
+            -  litmuschaos/argoexec:v2.11.0
```
Comment (gdsoumya) [inline charts/litmus-agent/charts/workflow-controller/templates/deployment.yaml:40]: Possibly want this to be a variable

File: charts/litmus-agent/charts/workflow-controller/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            - --container-runtime-executor
+            -  k8sapi
```
Comment (gdsoumya) [inline charts/litmus-agent/charts/workflow-controller/templates/deployment.yaml:42]: for the matter these are also configurable so best to kee the args as variable input and have default values.

**Review by gdsoumya — CHANGES_REQUESTED** [review]

File: charts/litmus-agent/charts/workflow-controller/templates/deployment.yaml (hunk 2 above)
Comment (Vr00mm) [reply charts/litmus-agent/charts/workflow-controller/templates/deployment.yaml]: Done

File: charts/litmus-agent/charts/workflow-controller/templates/deployment.yaml (hunk 3 above)
Comment (Vr00mm) [reply charts/litmus-agent/charts/workflow-controller/templates/deployment.yaml]: Done

**Review by gdsoumya — APPROVED** [review]

File: charts/litmus-agent/templates/hook-pre-install-job.yaml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        app.kubernetes.io/instance: {{ .Release.Name | quote }}
+        helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    spec:
+      restartPolicy: Never
+      serviceAccountName: {{ include "litmus-agent.serviceAccountName" . }}
+      containers:
+      - name: {{ .Chart.Name }}
+        imagePullPolicy: Always
+        #image: vr00mm/litmus-agent:latest
+        image: "vr00mm/litmus-agent-registrer:0.6.0"
```
Comment (rajdas98) [inline charts/litmus-agent/templates/hook-pre-install-job.yaml:29]: @Vr00mm We need to change the image registry

**Pull request closed without merging**